components/ai_chat/browser/ai_chat_metrics.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/browser/ai_chat_metrics.h"
+
+#include <algorithm>
+
+#include "base/metrics/histogram_macros.h"
+#include "brave/components/ai_chat/common/pref_names.h"
+#include "brave/components/p3a_utils/bucket.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+
+namespace ai_chat {
+
+namespace {
+constexpr base::TimeDelta kReportInterval = base::Hours(24);
+constexpr base::TimeDelta kReportDebounceDelay = base::Seconds(3);
+const int kChatCountBuckets[] = {1, 5, 10, 20, 50};
+const int kAvgPromptCountBuckets[] = {2, 5, 10, 20};
+}  // namespace
+
+const char kChatCountHistogramName[] = "Brave.AIChat.ChatCount";
+const char kAvgPromptCountHistogramName[] = "Brave.AIChat.AvgPromptCount";
+const char kEnabledHistogramName[] = "Brave.AIChat.Enabled";
+const char kUsageDailyHistogramName[] = "Brave.AIChat.UsageDaily";
+
+AIChatMetrics::AIChatMetrics(PrefService* local_state)
+    : chat_count_storage_(local_state,
+                          prefs::kBraveChatP3AChatCountWeeklyStorage),
+      prompt_count_storage_(local_state,
+                            prefs::kBraveChatP3APromptCountWeeklyStorage) {
+  ReportCounts();
+}
+
+AIChatMetrics::~AIChatMetrics() = default;
+
+void AIChatMetrics::RegisterPrefs(PrefRegistrySimple* registry) {
+  registry->RegisterListPref(prefs::kBraveChatP3AChatCountWeeklyStorage);
+  registry->RegisterListPref(prefs::kBraveChatP3APromptCountWeeklyStorage);
+}
+
+void AIChatMetrics::RecordEnabled() {
+  UMA_HISTOGRAM_BOOLEAN(kEnabledHistogramName, true);
```

Comment:
we do not want to emit false signal by default? So we are going to collect only `true`s and then look at the amount?

Reply:
Yes, that's right. We will use the total samples from another question as the denominator for calculating the ratio. More details on this process can be found in the handbook that I wrote.